File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/atlas/compass/CompassUpdatePartyMemberMessage.scala





















// Generated on 05/03/2013 10:34:35
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait CompassUpdatePartyMemberMessage extends CompassUpdateMessage {
  override val messageId = 5589

  protected trait Message extends CompassUpdateMessage.Message with Serializable {
    override val messageId = CompassUpdatePartyMemberMessage.messageId

    val memberId: Int

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeInt(memberId)
    }

  }

  protected def create(theType: Byte, worldX: Short, worldY: Short, memberId: Int): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val theType = buf.readByte()
    val worldX = buf.readShort()
    val worldY = buf.readShort()
    val memberId = buf.readInt()

    create(theType, worldX, worldY, memberId)
  }

}

object CompassUpdatePartyMemberMessage extends CompassUpdatePartyMemberMessage {
  case class MessageImpl(theType: Byte, worldX: Short, worldY: Short, memberId: Int) extends Message
  type Target = MessageImpl

  override def create(theType: Byte, worldX: Short, worldY: Short, memberId: Int) = MessageImpl(theType, worldX, worldY, memberId)

  def apply(theType: Byte, worldX: Short, worldY: Short, memberId: Int) = create(theType, worldX, worldY, memberId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/CompassUpdatePartyMemberMessage.java


// Generated on 05/08/2013 19:37:40
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class CompassUpdatePartyMemberMessage extends CompassUpdateMessage {
    public static final int MESSAGE_ID = 5589;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int memberId;
    
    public CompassUpdatePartyMemberMessage() { }
    
    public CompassUpdatePartyMemberMessage(byte type, short worldX, short worldY, int memberId) {
        super(type, worldX, worldY);
        this.memberId = memberId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeInt(memberId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        memberId = buf.readInt();
        if (memberId < 0)
            throw new RuntimeException("Forbidden value on memberId = " + memberId + ", it doesn't respect the following condition : memberId < 0");
    }
    
}
